catalyst/dl/utils/trace.py

```diff
@@ -125,14 +138,161 @@ def trace_model(
     getattr(model, mode)()
     set_requires_grad(model, requires_grad=requires_grad)
 
-    _runner_model, _runner_device = runner.model, runner.device
+    predict_fn(tracer, batch, **params)
+
+    return tracer.tracing_result
+
+
+def trace_model_from_checkpoint(
+    logdir: Path,
+    method_name: str,
+    checkpoint_name: str,
+    stage: str = None,
+    loader: Union[str, int] = None,
+    mode: str = "eval",
+    requires_grad: bool = False,
+    opt_level: str = None,
+    device: Device = "cpu",
+):
+    """
+    Traces model using created experiment and runner.
+
+    Args:
+        logdir (Union[str, Path]): Path to Catalyst logdir with model
+        checkpoint_name (str): Name of model checkpoint to use
+        stage (str): experiment's stage name
+        loader (Union[str, int]): experiment's loader name or its index
+        method_name (str): Model's method name that will be
+            used as entrypoint during tracing
+        mode (str): Mode for model to trace (``train`` or ``eval``)
+        requires_grad (bool): Flag to use grads
+        opt_level (str): AMP FP16 init level
+        device (str): Torch device
+
+    Returns:
+        the traced model
+    """
+    config_path = logdir / "configs" / "_config.json"
+    checkpoint_path = logdir / "checkpoints" / f"{checkpoint_name}.pth"
+    print("Load config")
+    config: Dict[str, dict] = load_config(config_path)
+    runner_params = config.get("runner_params", {}) or {}
+
+    # Get expdir name
+    config_expdir = Path(config["args"]["expdir"])
+    # We will use copy of expdir from logs for reproducibility
+    expdir = Path(logdir) / "code" / config_expdir.name
+
+    print("Import experiment and runner from logdir")
+    ExperimentType, RunnerType = import_experiment_and_runner(expdir)
+    experiment: Experiment = ExperimentType(config)
```

Comment:
due to `Experiment` and `Runner` usage, it's better to move them to `catalyst.core.utils`

Reply:
move `trace_model_from_checkpoint` function :)

Reply:
ok, will do

Reply:
Done!